### PR TITLE
Feature/oc 13122 - NavTabs Design Tokens

### PR DESCRIPTION
**Pull request by kukoc**
**Checklist**
_(Check off all the items before submitting)_
- [x] Build process is done without errors and all tests pass in `/lib` directory.
- [x] Self-reviewed the code prior to submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed. -> Specs about NavTab tokens will be added in a separate Jira 
- [x] Added/updated tests as needed. -> Not needed

**Purpose**
Adding Design Tokens to the NavTabs Component implementation

### CODE REVIEW

File: lib/src/common/variables.js [hunk 1]
```diff
@@ -12,7 +12,10 @@ export const globalTokens = {
   hal_grey_s_40: "#666666",
   hal_black: "#000000",
   color_grey_800: "#4d4d4d",
+  color_grey_700: "#666666",
   color_grey_600: "#808080",
```
Comment (raquelarrojo) [inline lib/src/common/variables.js:16]: `#666666` already exists in `hal_grey_s_40`.

File: lib/src/common/variables.js [hunk 2]
```diff
@@ -12,7 +12,10 @@ export const globalTokens = {
...
+  color_grey_500: "#999999",
+
```
Comment (raquelarrojo) [inline lib/src/common/variables.js:18]: `#999999` already exists in `hal_grey_l_60`.

File: lib/src/common/variables.js [hunk 3]
```diff
@@ -26,8 +29,9 @@ export const globalTokens = {
   hal_purple_s_38: "#5f249f",
   hal_purple_d_30: "#4b1c7d",
   hal_purple_d_20: "#321353",
-  hal_purple_d_70: "#9a6bb2",
-  color_purple_600: "#7d2fd0",
+  hal_purple_d_70: "#9A6BB2",
+  color_purple_600: "#7D2FD0",
```
Comment (raquelarrojo) [inline lib/src/common/variables.js:33]: Please keep `#9a6bb2` and `#7d2fd0` in lowercase to keep it as in the rest of the components.

File: lib/src/common/variables.js [hunk 4]
```diff
@@ -26,8 +29,9 @@ export const globalTokens = {
...
+  color_purple_700: "#5F249F",
   color_purple_300: "#cbacec",
```
Comment (raquelarrojo) [inline lib/src/common/variables.js:35]: `#5f249f` already exists in `hal_purple_s_38`.

File: lib/src/common/variables.js [hunk 5]
```diff
@@ -37,7 +41,9 @@ export const globalTokens = {
   hal_blue_s_35: "#0067b3",
   hal_blue_d_20: "#003c66",
   color_blue_200: "#cceaff",
-  color_blue_500: "#33aaff",
+
+  color_blue_500: "#33AAFF",
+  color_blue_600: "#0095FF",
   color_blue_50: "#f5fbff",
```
Comment (raquelarrojo) [inline lib/src/common/variables.js:47]: Please keep `#33aaff` in lowercase.
`#0095ff` already exists in `hal_blue_l_50`.

File: website/screens/theme-generator/themes/opinionated-theme.json [hunk 6]
```diff
@@ -64,6 +64,9 @@
   "link": {
     "baseColor": "#5f249f"
   },
+  "navTabs": {
+    "baseColor": "#5f249f"
+  },
```
Comment (raquelarrojo) [inline website/screens/theme-generator/themes/opinionated-theme.json:69]: I would remove this until the opinionated theme is done.

File: website/screens/theme-generator/themes/schemas/opinionated.schema.json [hunk 7]
```diff
@@ -64,6 +64,9 @@
   "link": {
     "baseColor": "color"
   },
+  "navTabs": {
+    "baseColor": "color"
+  },
```
Comment (raquelarrojo) [inline website/screens/theme-generator/themes/schemas/opinionated.schema.json:69]: I would remove this until the opinionated theme is done.

File: website/screens/components/tabs/specs/TabsSpecsPage.tsx [hunk 8]
```diff
@@ -266,6 +266,53 @@ const sections = [
...
+              <tr>
+                <td>
+                  <Code>titleLabelFontSize</Code>
+                </td>
+                <td>Title</td>
+                <td>
+                  <Code>font-scale-03</Code>
+                </td>
+                <td>16px</td>
+              </tr>
```
Comment (raquelarrojo) [inline website/screens/components/tabs/specs/TabsSpecsPage.tsx:301]: Please change this to `1rem / 16px`.

File: website/screens/components/tabs/specs/TabsSpecsPage.tsx [hunk 9]
```diff
@@ -266,6 +266,53 @@ const sections = [
...
+                  <Code>titleLabelFontWeight</Code>
+                </td>
+                <td>Title</td>
+                <td>
+                  <Code>font-regular</Code>
+                </td>
+                <td>600</td>
+              </tr>
+            </tbody>
+          </DxcTable>
```
Comment (raquelarrojo) [inline website/screens/components/tabs/specs/TabsSpecsPage.tsx:313]: Pleas change tokens to `fontFamily`, `fontSize` and `fontWeight`.
Add `fontStyle` token to the table.

File: lib/src/tabs-nav/Tab.tsx [hunk 10]
```diff
@@ -103,16 +117,18 @@ const Tab = styled.a<TabStyleProps>`
...
+    outline: 2px solid ${props.theme.focusOutline};
   }
 
   :active {
-    background: #0000001A;
+    background: ${props.theme.pressedBackgroundColor};
     outline: 2px solid #33aaff};
+    color: ${props.theme.selectedFontColor};
+
   }`}
```
Comment (raquelarrojo) [inline lib/src/tabs-nav/Tab.tsx:132]: Remove unused `LabelContainer` styled component.

File: lib/src/tabs-nav/Tab.tsx [hunk 11]
```diff
@@ -103,16 +117,18 @@ const Tab = styled.a<TabStyleProps>`
   ${(props) =>
     !props.disabled &&
     `:hover {
-    background: #0000000D;
+    background: ${props.theme.unselectedBackgroundColor};
   }
```
Comment (raquelarrojo) [inline lib/src/tabs-nav/Tab.tsx:121]: Change this token to `hoverBackgroundColor`.

File: lib/src/tabs-nav/Tab.tsx [hunk 12]
```diff
@@ -103,16 +117,18 @@ const Tab = styled.a<TabStyleProps>`
...
-    outline: 2px solid #33aaff};
+    outline: 2px solid ${props.theme.focusOutline};
   }
 
   :active {
-    background: #0000001A;
+    background: ${props.theme.pressedBackgroundColor};
     outline: 2px solid #33aaff};
+    color: ${props.theme.selectedFontColor};
+
```
Comment (raquelarrojo) [inline lib/src/tabs-nav/Tab.tsx:131]: This `color` sentence is not needed, since it is applied in the `DxcTypography`.

**Review by raquelarrojo — CHANGES_REQUESTED** [review]
I would also add tokens for icon in NavTabs component, as in Tabs component.
Now, there are no styles applied to the icon.

Notification badge should be aligned in row with the label.
Now:
image

Before:
image

File: lib/src/common/variables.js (hunk 1 above)
Comment (kukoc) [reply lib/src/common/variables.js:16]: ok, i'll chage it. As we discuss, names are quite confusing

File: lib/src/common/variables.js (hunk 2 above)
Comment (kukoc) [reply lib/src/common/variables.js:18]: done

File: lib/src/common/variables.js (hunk 3 above)
Comment (kukoc) [reply lib/src/common/variables.js:33]: done

File: lib/src/common/variables.js (hunk 4 above)
Comment (kukoc) [reply lib/src/common/variables.js:35]: done

File: lib/src/common/variables.js (hunk 5 above)
Comment (kukoc) [reply lib/src/common/variables.js:47]: done

File: website/screens/components/tabs/specs/TabsSpecsPage.tsx (hunk 8 above)
Comment (kukoc) [reply website/screens/components/tabs/specs/TabsSpecsPage.tsx:301]: done

File: website/screens/components/tabs/specs/TabsSpecsPage.tsx (hunk 9 above)
Comment (kukoc) [reply website/screens/components/tabs/specs/TabsSpecsPage.tsx:313]: done

File: lib/src/tabs-nav/Tab.tsx (hunk 10 above)
Comment (kukoc) [reply lib/src/tabs-nav/Tab.tsx:135]: I'm using it now no align the label and the notifications  using a flex container with row direction

File: lib/src/tabs-nav/Tab.tsx (hunk 11 above)
Comment (kukoc) [reply lib/src/tabs-nav/Tab.tsx:124]: done

File: lib/src/tabs-nav/Tab.tsx (hunk 12 above)
Comment (kukoc) [reply lib/src/tabs-nav/Tab.tsx:134]: done

File: lib/src/common/variables.js [hunk 13]
```diff
@@ -680,6 +683,25 @@ export const componentTokens = {
...
+    focusOutline: globalTokens.hal_blue_l_50,
+    selectedUnderlineColor: globalTokens.hal_purple_s_38,
+    dividerColor: globalTokens.hal_grey_l_75,
+    fontFamily: globalTokens.type_sans,
+    fontSize: globalTokens.type_scale_03,
+    fontStyle: globalTokens.type_normal,
+    fontWeight: globalTokens.type_regular,
+    selectedIconColor: globalTokens.color_grey_700,
+    unselectedIconColor: globalTokens.color_grey_700,
+    disabledIconColor: globalTokens.hal_grey_l_60
```
Comment (raquelarrojo) [inline lib/src/common/variables.js:703]: `selectedIconColor` and `unselectedIconColor` have same value (`#666666`). They should have different value I think.

File: lib/src/tabs-nav/Tab.tsx [hunk 14]
```diff
@@ -128,6 +145,19 @@ const TabIconContainer = styled.div<TabIconContainerProps>`
...
+    svg {
+      color: ${(props) => props.theme.selectedIconColor};
+    }
+  }
+
+  &[aria-disabled="true"] {
+    svg {
+      color: ${(props) => props.theme.disabledIconColor};
+    }
   }
```
Comment (raquelarrojo) [inline lib/src/tabs-nav/Tab.tsx:161]: Disabled and selected styles are not being applied to the icon. Please change these styles to `TabIconContainer` styled component.

```
&[aria-selected="true"] {
    svg {
      color: ${(props) => props.theme.selectedIconColor};
    }
  }

  &[aria-disabled="true"] {
    svg {
      color: ${(props) => props.theme.disabledIconColor};
    }
  }
```

Also, `aria-disabled` is not present to the Tab, so it should be added to the TabContainer as `aria-disabled={disabled}`.

**Review by raquelarrojo — CHANGES_REQUESTED** [review]

File: website/screens/theme-generator/themes/opinionated-theme.json (hunk 6 above)
Comment (kukoc) [reply website/screens/theme-generator/themes/opinionated-theme.json:69]: removed!

File: website/screens/theme-generator/themes/schemas/opinionated.schema.json (hunk 7 above)
Comment (kukoc) [reply website/screens/theme-generator/themes/schemas/opinionated.schema.json:69]: removed!

**Review by raquelarrojo — APPROVED** [review]

**Pull request merged**